src/main/java/org/folio/circulation/domain/MoveRequestService.java

```diff
@@ -36,28 +41,35 @@ public MoveRequestService(RequestRepository requestRepository,
     this.requestLoanValidator = requestLoanValidator;
     this.requestNoticeSender = requestNoticeSender;
     this.configurationRepository = configurationRepository;
+    this.eventPublisher = eventPublisher;
   }
 
-  public CompletableFuture<Result<RequestAndRelatedRecords>> moveRequest(
-      RequestAndRelatedRecords requestAndRelatedRecords) {
-    return completedFuture(of(() -> requestAndRelatedRecords))
-      .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::findDestinationItem))
-      .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::getDestinationRequestQueue))
-      .thenApply(r -> r.map(this::pagedRequestIfDestinationItemAvailable))
-      .thenCompose(r -> r.after(this::validateUpdateRequest))
-      .thenComposeAsync(r -> r.combineAfter(configurationRepository::findTimeZoneConfiguration,
-        RequestAndRelatedRecords::withTimeZone))
-      .thenCompose(r -> r.after(updateUponRequest.updateRequestQueue::onMovedTo))
-      .thenComposeAsync(r -> r.after(this::updateRelatedObjects))
-      .thenCompose(r -> r.after(requestRepository::update))
-      .thenApply(r -> r.next(requestNoticeSender::sendNoticeOnRequestMoved))
-      .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::findSourceItem))
-      .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::getSourceRequestQueue))
-      .thenCompose(r -> r.after(updateUponRequest.updateRequestQueue::onMovedFrom))
-      .thenComposeAsync(r -> r.after(this::updateRelatedObjects))
-      .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::findDestinationItem))
-      .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::getDestinationRequestQueue))
-      .thenComposeAsync(r -> r.after(moveRequestProcessAdapter::getRequest));
+  public CompletableFuture<Result<RequestAndRelatedRecords>> moveRequest(RequestAndRelatedRecords requestAndRelatedRecords) {
+
+    return moveRequestProcessAdapter.getRequest(requestAndRelatedRecords)
```

Comment:
Could you please store the original request in the `RequestAndRelatedRecords` context object or create an overload method:
` moveRequest(RequestAndRelatedRecords context, Request oldRequest)`

So, there will be no nested blocks.